### ISSUE TITLE
Updating item name onDrawerItemClickListener

### ISSUE THREAD

**Issue by YPierru** [mikepenz/MaterialDrawer#1640]
I'm trying to update the name of one of my item like that : 

```java
    .withOnDrawerItemClickListener(new Drawer.OnDrawerItemClickListener() {
                    @Override
                    public boolean onItemClick(View view, int position, IDrawerItem drawerItem) {
                        if (drawerItem instanceof Nameable && drawerItem != null) {

                            Nameable nameable = (Nameable)drawerItem;

                            if(drawerItem.getIdentifier()==Constants.ID_DL_ITEM_ALERT_TEXT){
                                String str = "new name";
                                PrimaryDrawerItem pdi = (PrimaryDrawerItem) drawerItem;
                                pdi.withName(str);
                                result.updateItem(pdi);
                            }

                        return false;
                    }
                })
```

The new String is not this one but it's just an example. I did not paste the complete code of my drawer, of course the item I'm retrieving is a PrimaryDrawerItem.
The name is not updated until I click a second time on the item... Do you know why ?
Thanks.

**Comment by YPierru** [comment]
@mikepenz No problem.
I'm reassured that my code is fine. Thanks for your answer, let me know if I can help you by any means.

**Comment by mikepenz** [comment]
@YPierru just tried by adding 

```java
if (drawerItem instanceof Nameable && drawerItem != null) {

    Nameable nameable = (Nameable) drawerItem;

    if (drawerItem.getIdentifier() == 1) {
        String str = "new name";
        nameable.withName(str);
        result.updateItem(drawerItem);
    }

    return true;
}
```

before the general intent logic in the `DrawerActivity` of the sample. And it works without any issues. 
have you debugged through? is it hitting this code? How is the click triggered. Through normal touch in the app or programatically? 

which android phone do you use? which android version?

**Comment by YPierru** [comment]
@mikepenz Okay I found where was my mistake... The code I gave you actually works fine. I simplified it relative to the exact code in my app. The exact code retrieve the "new string" from an AlertDialog with an EditText. But because the AlertDialog is asynchronous the code following the call is executed without the new string.... That's why the new string appears only when I click again on the Item.

Now I only call the AlertDialog in the onDrawerClickListener. The item label is updated in the onDrawerOpen event. (I have many AlertDialog which are in a separate class so I can't update my drawer item in the alertdialog button listener).

However thank you for you help, and sorry to bother you.

Yan.
